### DISCUSSION TITLE
[Web] Add support for playing Android Motion Photos

### DISCUSSION THREAD

**Discussion by ua741** [ente-io/ente#561] [Enhancements]
Motion photos that are uploaded via ente photos app or imported from web app from any source apart from Google Photos export are not playable on the web/desktop app.

**How to identify the Android Motion Photo:** 
FileType will be image and it will contain `mvi` key in the publicMetadata and the value of the field should be > 0.

**How to extract video of Android Motion photo**: 
The video is part of the image blob. We can read the video by reading the file Bytes from mvi index to end.

**Comment by shuoO-24** [comment]
Hi @ua741 , I would like to work on this issue. Can you give some directions about where I can start with? Thanks very much for your help!

**Comment by abhinavkgrd** [comment]
Hey @shuoO-24, thanks!

We currently have a [decodeLivePhoto function](https://github.com/ente-io/photos-web/blob/cb0cc11058fde5368e0a13bc713b6ffafbfbb4e6/apps/photos/src/services/livePhotoService.ts#L15) used by different parts of the app to decode a downloaded Apple Live photo zip. 

we can extend that itself to include the identification check and decode the motion photo accordingly.

As, for the implementation of identification checks,

`mvi` is a new key, that was not used earlier on the web, so we will need to add it to [FilePulbicMagicMetadataProps](https://github.com/ente-io/photos-web/blob/cb0cc11058fde5368e0a13bc713b6ffafbfbb4e6/apps/photos/src/types/file/index.ts#L93).

 And the checks for detection are  
 if file is a live photo-> `file.metadata.fileType === FILE_TYPE.LIVE_PHOTO`
 and for motion photo-> `file.pubMagicMetadata.data.mvi >0 `

**Comment by FranckRJ** [comment]
Hello, just to be sure, does it means that Motion Photos are still properly uploaded to you servers or once this bug is fixed I will have to upload all Motion Photos again so I can see them on ente desktop ?

**Reply by ua741** [reply]
> Hello, just to be sure, does it means that Motion Photos are still properly uploaded to you servers or once this bug is fixed I will have to upload all Motion Photos again so I can see them on ente desktop ?

Android Motion photos are uploaded properly, we just need to add playback support on Web/Desktop.

**Comment by ua741** [comment]
Related Issues
[Google Top Shot are not recognized as such when uploading from the desktop app](https://github.com/ente-io/ente/issues/1177)
[Collect link: Motion Photos are uploaded as still images only](https://github.com/ente-io/ente/issues/10527)
